BuildingBlocks/src/DevelopmentKit.Identity/ValueObjects/IdentityAddress.cs

```diff
@@ -45,37 +47,38 @@ public static bool IsValid(string? stringValue)
 
         var lengthIsValid = stringValue.Length <= MAX_LENGTH;
 
-        var realm = stringValue[..3];
+        const string pattern = @"^did\:e\:((?:[a-zA-Z]+\.)+[a-zA-Z]+)\:dids\:(.+)(.{2})$";
```

Comment:
This regex will be used quite often. So it probably makes sense to convert it into a compiled one. ReSharper should suggest you to convert it for you, and it will give you something like this:

```cs
    [GeneratedRegex(@"^did\:e\:((?:[a-zA-Z]+\.)+[a-zA-Z]+)\:dids\:(.+)(.{2})$", RegexOptions.IgnoreCase, "de-DE")]
    private static partial Regex AddressValidationRegex();
```

which you can then use as follows:

```cs
var matches = AddressValidationRegex().Matches(stringValue).First().Groups;
```

Reply:
This turns `public record IdentityAddress` into a `public partial record IdentityAddress`. Is that acceptable?

Reply:
Of course. Why not?

Reply:
For no specific reason, I'm just not sure what ramifications that may have.